mayastor/src/subsys/opts.rs

```diff
@@ -482,15 +482,36 @@ impl GetOpts for IscsiTgtOpts {
 pub struct ErrStoreOpts {
     /// ring buffer size
     pub err_store_size: usize,
+
     /// NexusErrStore enabled
     pub enable_err_store: bool,
+
+    /// whether to fault the child due to failed re-tried IOs
+    pub fault_child_on_retry_count: bool,
+
+    /// the maximum number of errors for the same retried IO
+    pub max_retry_errors: u32,
+
+    /// whether to fault the child due to the total number of failed IOs
+    pub fault_child_on_error_count: bool,
+
+    /// the maximum number of errors in total
+    pub max_total_errors: u32,
+
+    /// errors older than this are ignored
+    pub max_error_age_ns: u64,
 }
 
 impl Default for ErrStoreOpts {
     fn default() -> Self {
         Self {
             err_store_size: 256,
-            enable_err_store: true,
+            enable_err_store: false,
+            fault_child_on_retry_count: false,
+            fault_child_on_error_count: false,
+            max_retry_errors: 0,
```

Comment:
is it a good idea to set these to 0? one retry and your out?

Reply:
I'd expect this to be set to some non-zero value if the feature is enabled but I didn't want to assign an arbitrary number. 
On the other hand, a common retry value in SPDK is 5, which seems reasonable, so I'll set that as the default.